src/platform/plugins/shared/esql/public/triggers/esql_controls/control_flyout/identifier_control_form.tsx

```diff
@@ -78,6 +81,7 @@ export function IdentifierControlForm({
         const queryForFields = getQueryForFields(queryString, cursorPosition);
         getESQLQueryColumnsRaw({
           esqlQuery: queryForFields,
+          variables: esqlVariables,
```

Comment:
This is fixing the first bug, correctly run the query for fields when there is an existing variable in the query